### PR TITLE
feature: add dnsPolicy to fluentbit, more options to kubernetes filter

### LINKED ISSUE #524: add dnsPolicy support to the FluentBit custom resource
### Is your feature request related to a problem? Please describe.

I'd like to use the `use_kubelet` option when configuring the kubernetes fliter plugin.  As noted in the [documentation](https://docs.fluentbit.io/manual/pipeline/filters/kubernetes#configuration-setup) you need to have a daemonset that uses 

```
hostNetwork: true
dnsPolicy: ClusterFirstWithHostNet
```

### Describe the solution you'd like

This request is to add support for `dnsPolicy` in the FluentBit custom resource.   That would let us create a FluentBit resource similar to this truncated version

```
apiVersion: fluentbit.fluent.io/v1alpha2
kind: FluentBit
metadata:
  name: fluent-bit
  labels:
    app.kubernetes.io/name: fluent-bit
spec:
  volumes:
  - name: journal
    hostPath:
      path: /var/log/journal
  - name: mnt
    hostPath:
      path: /mnt
  - name: varlibdockercontainers
    hostPath:
      path: /var/snap/microk8s/
  - name: varlog
    hostPath:
      path: /var/log
  tolerations:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
    operator: Exists
  hostNetwork: true
  dnsPolicy: ClusterFirstWithHostNet
.
.
.
```

### Additional context

This would support https://github.com/fluent/fluent-operator/issues/514 but I think this could be done independently b/c if dnsPolicy is added then people could add a ClusterFilter with the customPlugin feature to include the kubernetes filter plugin

### PR DESCRIPTION
**Pull request by jjsiv**
Signed-off-by: Jedrzej Kotkowski <jedrzejk143@gmail.com>

### What this PR does / why we need it:
- Adds dnsPolicy option to FluentBit CRD
- Adds these options to FluentBit k8s filter plugin: dnsRetries, dnsWaitTime, useKubelet, kubeletPort, kubeletHost, kubeMetaCacheTTL, kubeTokenTTL
### Which issue(s) this PR fixes:

Fixes #524, #514

### Does this PR introduced a user-facing change?

```release-note
Users can now specify dnsPolicy for FluentBit as well as more options for Kubernetes filter plugin
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by benjaminhuo** [comment]
@jjsiv Thanks for this PR, it's a big enhancement!
Would you please run `make docs-update` to update the API docs?

File: apis/fluentbit/v1alpha2/plugins/filter/kubernetes_types.go [hunk 1]
```diff
@@ -65,6 +67,26 @@ type Kubernetes struct {
...
+	// kubelet host using for HTTP request, this only works when Use_Kubelet set to On.
+	KubeletHost string `json:"kubeletHost,omitempty"`
+	// configurable TTL for K8s cached metadata. By default, it is set to 0
+	// which means TTL for cache entries is disabled and cache entries are evicted at random
+	// when capacity is reached. In order to enable this option, you should set the number to a time interval.
+	// For example, set this value to 60 or 60s and cache entries which have been created more than 60s will be evicted.
+	KubeMetaCacheTTL string `json:"kubeMetaCacheTTL,omitempty"`
+	// configurable 'time to live' for the K8s token. By default, it is set to 600 seconds.
+	// After this time, the token is reloaded from Kube_Token_File or the Kube_Token_Command.
+	KubeTokenTTL *int32 `json:"kubeTokenTTL,omitempty"`
```
Comment (wanjunlei) [inline apis/fluentbit/v1alpha2/plugins/filter/kubernetes_types.go:89]: The `Kube_Token_TTL` is a time interval, so it should be `string`.

**Review by wanjunlei — CHANGES_REQUESTED** [review]

File: apis/fluentbit/v1alpha2/plugins/filter/kubernetes_types.go (hunk 1 above)
Comment (jjsiv) [reply apis/fluentbit/v1alpha2/plugins/filter/kubernetes_types.go]: Thanks, fixed

**Comment by jjsiv** [comment]
@benjaminhuo for whatever reason running docs-update didn't actually update the docs for kubernetes filter, only a bunch of other stuff that I didn't include in the commit. I added it manually, maybe it would be worth having a look at later

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
> @benjaminhuo for whatever reason running docs-update didn't actually update the docs for kubernetes filter, only a bunch of other stuff that I didn't include in the commit. I added it manually, maybe it would be worth having a look at later

Great, thank you!

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**